src/gui/tray/UnifiedSearchResultItem.qml

```diff
@@ -0,0 +1,232 @@
+                sourceSize.width: imageData.width
+                sourceSize.height: imageData.height
+                width: imageData.width
+                height: imageData.height
+            }
+            Rectangle {
+                id: mask
+                visible: false
+                radius: model.isRounded ? width / 2 : 0
+                width: imageData.width
+                height: imageData.height
+            }
+            OpacityMask {
+                id: imageData
+                visible: !unifiedSearchResultThumbnailPlaceholder.visible
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                Layout.leftMargin: iconLeftMargin
+                Layout.preferredWidth: model.icons ? unifiedSearchResultImageContainer.iconWidth : 0
+                Layout.preferredHeight: model.icons ? unifiedSearchResultImageContainer.iconWidth: 0
+                source: unifiedSearchResultThumbnail
+                maskSource: mask
+            }
+            Image {
+                id: unifiedSearchResultThumbnailPlaceholder
+                visible: model.icons && model.imagePlaceholder && unifiedSearchResultThumbnail.status != Image.Ready
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                Layout.leftMargin: iconLeftMargin
+                verticalAlignment: Qt.AlignCenter
+                cache: true
+                source: model.imagePlaceholder
+                sourceSize.height: unifiedSearchResultImageContainer.iconWidth
+                sourceSize.width: unifiedSearchResultImageContainer.iconWidth
+                Layout.preferredWidth: visible ? unifiedSearchResultImageContainer.iconWidth : 0
+                Layout.preferredHeight: visible ? unifiedSearchResultImageContainer.iconWidth : 0
+            }
+        }
+
+        ColumnLayout {
+            id: unifiedSearchResultTextContainer
+            Layout.fillWidth: true
+
+            Text {
+                id: unifiedSearchResultTitleText
+                text: model.resultTitle.replace(/[\r\n]+/g, " ")
+                visible: parent.visible
+                Layout.leftMargin: textLeftMargin
+                Layout.rightMargin: textRightMargin
+                Layout.fillWidth: true
+                elide: Text.ElideRight
+                font.pixelSize: Style.topLinePixelSize
+                color: "black"
+            }
+            Text {
+                id: unifiedSearchResultTextSubline
+                text: model.subline.replace(/[\r\n]+/g, " ")
+                elide: Text.ElideRight
+                visible: parent.visible
+                Layout.leftMargin: textLeftMargin
+                Layout.rightMargin: textRightMargin
+                Layout.fillWidth: true
+                color: "grey"
+            }
+        }
+
+    }
+
+    RowLayout {
+        id: unifiedSearchResultItemFetchMore
+        visible: isFetchMoreTrigger
+
+        width: visible ? unifiedSearchResultMouseArea.width : 0
```

Comment:
no need to change size when item is not visible
when an item is not visible, it takes no space